sktime/performance_metrics/tests/test_performance_metrics_forecasting.py

```diff
@@ -2,64 +2,524 @@
+            else:
+                function_loss = loss_func(y_true, y_pred, y_train)
+                class_loss = loss_class.fn(y_true, y_pred, y_train)
+        elif "y_pred_benchmark" in loss_func_args:
+            if len(class_attrs) > 0:
+                function_loss = loss_func(
+                    y_true, y_pred, y_pred_benchmark, **class_attrs
+                )
+                class_loss = loss_class.fn(
+                    y_true, y_pred, y_pred_benchmark, **class_attrs
+                )
+            else:
+                function_loss = loss_func(y_true, y_pred, y_pred_benchmark)
+                class_loss = loss_class.fn(y_true, y_pred, y_pred_benchmark)
+        else:
+            if len(class_attrs) > 0:
+                function_loss = loss_func(y_true, y_pred, **class_attrs)
+                class_loss = loss_class.fn(y_true, y_pred, **class_attrs)
+            else:
+                function_loss = loss_func(y_true, y_pred)
+                class_loss = loss_class.fn(y_true, y_pred)
 
-    y_pred = y_test * 1.000001
-    assert smape_loss(y_test, y_pred) == pytest.approx(0.000001)
-    assert smape_loss(y_pred, y_test) == pytest.approx(0.000001)
+        # Assertion for functions and class having same result
+        assert np.isclose(function_loss, class_loss), " ".join(
+            [
+                "Expected loss function and class to return equal values,",
+                f"but loss function {loss_func.__name__} returned {function_loss}",
+                f"and {loss_class.name} returned {class_loss}.",
+            ]
+        )
 
-    y_pred = y_test * 2.0
-    assert smape_loss(y_test, y_pred) == pytest.approx(0.6666666)
-    assert smape_loss(y_pred, y_test) == pytest.approx(0.6666666)
 
-    y_pred = y_test * 100
-    assert smape_loss(y_test, y_pred) == smape_loss(y_pred, y_test)
+@pytest.mark.parametrize("loss_func_name", LOSS_RESULTS.keys())
+def test_univariate_function_output_type(loss_func_name):
+    loss_func = LOSS_RESULTS[loss_func_name]["func"]
+    for random_seed in RANDOM_SEEDS:
+        y = _make_series(n_timepoints=75, random_state=random_seed)
+        y_train, y_true = y.iloc[:50], y.iloc[50:]
+        y_pred = y.shift(1).iloc[50:]
+        y_pred_benchmark = y.rolling(2).mean().iloc[50:]
+
+        loss_func_args = inspect.getfullargspec(loss_func).args
+
+        if "y_train" in loss_func_args:
+            function_loss = loss_func(y_true, y_pred, y_train)
+        elif "y_pred_benchmark" in loss_func_args:
+            function_loss = loss_func(y_true, y_pred, y_pred_benchmark)
+        else:
+            function_loss = loss_func(y_true, y_pred)
+
+        is_num = is_numeric_dtype(function_loss)
+        is_scalar = np.isscalar(function_loss)
+        assert is_num and is_scalar, " ".join(
+            ["Loss function with univariate input should return scalar number"]
+        )
+
+
+# @pytest.mark.parametrize("loss_func", LOSS_RESULTS.keys())
```

Comment:
why are they commented out? shall we remove them or uncomment them?

Reply:
Yea, we need to remove them. The use of the scikit-learn input validations rendered those moot since their functionality converts array-like things to arrays. I'll delete these in the last push I make.